### PR TITLE
Additional Configuration Parameters

### PR DESCRIPTION
**Pull request by PhilippSchmaelzle**
Adds the following parameters for configuration:
- ReverseX and Y
- Gamma
- PieceWiseLinearHDR
- DSPSubregion

Tested on ROS1 with AVT Mako223C

### CODE REVIEW

**Comment by icolwell-as** [comment]
Thanks for submitting! I'll try to test it out on our camera soon.

**Comment by PhilippSchmaelzle** [comment]
No problem!
You made some pretty cool updates from release 1.0.0 onward. Therefore I will no longer use my own fixed driver but start using this one and maybe try to contribute a little.

File: cfg/AvtVimbaCamera.cfg [hunk 1]
```diff
@@ -174,6 +179,11 @@ gen.add("exposure_auto_min",      int_t, 	  SensorLevels.RECONFIGURE_RUNNING, "T
 gen.add("exposure_auto_outliers", int_t,	  SensorLevels.RECONFIGURE_RUNNING, "The total pixels from top of the distribution that are ignored by the auto exposure algorithm (0.01% increments)", 0, 0, 10000)
 gen.add("exposure_auto_rate",     int_t, 	  SensorLevels.RECONFIGURE_RUNNING, "The rate at which the auto exposure function changes the exposure setting.100% is auto exposure adjustments running at full speed, and 50% is half speed.", 100, 1, 100)
 gen.add("exposure_auto_target",   int_t,    SensorLevels.RECONFIGURE_RUNNING, "The auto exposure target mean value as a percentage, from 0=black to 100=white.", 50, 0, 100)
+gen.add("exposure_mode",    	str_t,    SensorLevels.RECONFIGURE_RUNNING, "The control for exposure duration.", "Timed", edit_method = exposure_mode_enum)
```
Comment (icolwell-as) [inline cfg/AvtVimbaCamera.cfg]: Can you keep the spacing consistent with the rest of the file?

File: src/avt_vimba_camera.cpp [hunk 2]
```diff
@@ -948,6 +950,62 @@ void AvtVimbaCamera::updateExposureConfig(Config& config)
     configureFeature("ExposureAutoTarget", static_cast<VmbInt64_t>(config.exposure_auto_target),
                      config.exposure_auto_target);
   }
+  if (config.exposure_mode != config_.exposure_mode || on_init_) {
+      configureFeature("ExposureMode", config.exposure_mode ,config.exposure_mode);
```
Comment (icolwell-as) [inline src/avt_vimba_camera.cpp:954]: It looks like you might be using 4 spaces for indents instead of 2.

File: src/avt_vimba_camera.cpp [hunk 3]
```diff
@@ -948,6 +950,62 @@ void AvtVimbaCamera::updateExposureConfig(Config& config)
...
+  if (config.exposure_threshold_PWL1 != config_.exposure_threshold_PWL1 || on_init_) {
+      configureFeature("ThresholdPWL1",
+                      static_cast<VmbInt64_t>(config.exposure_threshold_PWL1), config.exposure_threshold_PWL1);
+  }
+}
+
+/** Change the Gamma configuration*/
+void AvtVimbaCamera::updateGammaConfig(Config& config)
+{
+    if (on_init_)
```
Comment (icolwell-as) [inline src/avt_vimba_camera.cpp]: Same here, stick to 2 spaces for indents please.

**Review by icolwell-as — CHANGES_REQUESTED** [review]
Just tested this today, works great! thanks for submitting!
Only some minor formatting things that need tweaking.

**Comment by PhilippSchmaelzle** [comment]
thx for the review! I fixed the indents

**Comment by PhilippSchmaelzle** [comment]
Sorry for the 1000 commits. I had to use GitHub Web interface and it was a mess with tabs and spaces...

**Review by icolwell-as — APPROVED** [review]
Thanks!

**Review by dvangeyn-as — APPROVED** [review]

**Pull request merged**